### PR TITLE
Add Drawer Navigator (Expo Router + React Navigation)

### PR DESCRIPTION
**Pull request by saimon24**
Allow Drawer navigation (side menu) next to tabs and stack layout for both Expo Router and React Navigation.

### CODE REVIEW

File: cli/src/templates/packages/expo-router/drawer/app/(drawer)/index.tsx.ejs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { View, Text } from 'react-native';
+
+const Page = () => {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/drawer/app/(drawer)/index.tsx.ejs:5]: Can you add a conditional here that changes this based on the `stylingPackage` selected? (stylesheet, nativewind, tamagui)

File: cli/src/templates/packages/expo-router/drawer/app/(drawer)/news.tsx.ejs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { View, Text } from 'react-native';
+
+const Page = () => {
+  return (
+    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
```
Comment (danstepanov) [inline cli/src/templates/packages/expo-router/drawer/app/(drawer)/news.tsx.ejs:5]: Also needs `stylingPackage` options

File: cli/src/types.ts [hunk 3]
```diff
@@ -15,10 +15,13 @@ export const availablePackages = [
 	'reactnavigation',
 	'supabase',
 	'stylesheet',
-	'tamagui'
+	'tamagui',
+	'@react-navigation/drawer',
```
Comment (danstepanov) [inline cli/src/types.ts:19]: want to sort these alphabetically?

File: cli/src/utilities/runCLI.ts [hunk 4]
```diff
@@ -72,6 +72,21 @@ export async function runCLI(toolbox: Toolbox): Promise<CliResults> {
 					type: navigationTypeSelect.toLowerCase() as NavigationTypes
 				}
 			});
+
+			if (navigationTypeSelect === 'Drawer') {
```
Comment (danstepanov) [inline cli/src/utilities/runCLI.ts:76]: you can remove this entire change, i think

**Review by danstepanov — CHANGES_REQUESTED** [review]
Almost ready!

**Comment by saimon24** [comment]
Updated code with your changes - just not sure why the block in runCLI can be removed, but it works without it 🤷‍♂️

**Review by danstepanov — APPROVED** [review]
LGTM!

**Pull request merged**